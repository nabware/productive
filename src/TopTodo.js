import React from "react";

import Todo from "./Todo";

/** Shows the top todo.
 *
 * Props:
 * - todos
 *
 * TodoApp -> TopTodo
 */

function TopTodo({ todos, update }) {
  if (todos.length === 0) return null;

  const incompleteTodos = todos.filter(t => !t.isComplete);
  if (incompleteTodos.length === 0) return null;
  // lowest-priority # is the highest priority
  let top = incompleteTodos.reduce(
    (acc, cur) => cur.priority < acc.priority ? cur : acc, incompleteTodos[0]);

  return <Todo todo={top} update={update} />;
}

export default TopTodo;