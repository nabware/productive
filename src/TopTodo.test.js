import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";
import TEST_TODOS from "./_testCommon.js";

describe("top todo", function () {
  it("renders without crashing", function () {
    render(<TopTodo todos={TEST_TODOS} />);
  });

  it("contains top priority todo", function () {
    const result = render(<TopTodo todos={TEST_TODOS} />);
    expect(result.queryByText("test1title")).toBeInTheDocument();
    expect(result.queryByText("test2title")).not.toBeInTheDocument();
    expect(result.queryByText("test3title")).not.toBeInTheDocument();
  });

  it ("snapshot", function () {
    const { container } = render(<TopTodo todos={TEST_TODOS} />);
    expect(container).toMatchSnapshot();
  });
});

