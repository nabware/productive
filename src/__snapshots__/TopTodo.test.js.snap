// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        test1title
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        test1description
      </small>
    </div>
  </div>
</div>
`;

exports[`top todo snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        test1title
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        test1description
      </small>
    </div>
  </div>
</div>
`;
